### PR TITLE
Add Slime Block

### PR DESCRIPTION
**Pull request by max-xoo**
I needed the slime blocks so I implemented them, it's client side everything that is slowing down and jumping

Here is a video to show that it works client side: [click here](https://www.youtube.com/watch?v=DC3cgPGDb5M&ab_channel=MaXoooZ)
(it's on df of course)

### CODE REVIEW

File: server/block/slime.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package block
+
+// Slime is a full-sized block of snow.
```
Comment (JustTalDevelops) [inline server/block/slime.go:3]: Looks like you forgot to change this?

File: server/block/hash.go [hunk 2]
```diff
@@ -136,6 +136,7 @@ const (
 	hashWoodStairs
 	hashWoodTrapdoor
 	hashWool
+	hashSlime
```
Comment (JustTalDevelops) [inline server/block/hash.go:139]: I would keep this list in alphabetical order - but you should probably be using the blockhash tool for this.

File: server/block/slime.go (hunk 1 above)
Comment (max-xoo) [reply server/block/slime.go:3]: Oh yeah, good point, I'm changing that

File: server/block/hash.go (hunk 2 above)
Comment (max-xoo) [reply server/block/hash.go:139]: Ok I'll change that too

File: server/block/slime.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package block
+
+// A Slime block is a translucent block that entities can bounce on
+type Slime struct {
+	solid
+}
+
+// BreakInfo ...
+func (s Slime) BreakInfo() BreakInfo {
+	return newBreakInfo(0.1, alwaysHarvestable, shovelEffective, oneOf(s))
```
Comment (mmm545) [inline server/block/slime.go:10]: i don't think there is any specific tool that's effective for breaking slimes?

Comment (max-xoo) [reply server/block/slime.go:10]: With the hand the block is one shot

File: server/block/slime.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package block
+
+// A Slime block is a translucent block that entities can bounce on
```
Comment (JustTalDevelops) [inline server/block/slime.go:3]: Just to stay consistent, I'd remove the `A ` prefix and add a period at the end of the sentence.

File: server/block/slime.go (hunk 3 above)
Comment (AndreasHGK) [reply server/block/slime.go:10]: yea, but the it says here the tool to harvest slime blocks is a shovel

File: server/block/slime.go (hunk 4 above)
Comment (max-xoo) [reply server/block/slime.go:3]: I do that

File: server/block/slime.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+type Slime struct {
```
Comment (T14Raptor) [inline server/block/slime.go:4]: This should be called `SlimeBlock` and the file should be renamed to `slime_block.go`

File: server/block/slime.go (hunk 4 above)
Comment (JustTalDevelops) [reply server/block/slime.go]: You might as well also remove the `block` suffix after `Slime`.

File: server/block/slime.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package block
+
+// Slime is a translucent block that entities can bounce on.
+type Slime struct {
```
Comment (T14Raptor) [inline server/block/slime.go:4]: This also should implement `block.LightDiffuser` with `LightDiffusionLevel()` returning the correct light diffusion level

File: server/block/slime.go (hunk 4 above)
Comment (max-xoo) [reply server/block/slime.go]: done

File: server/block/slime.go (hunk 3 above)
Comment (max-xoo) [reply server/block/slime.go]: done

File: server/block/slime.go (hunk 6 above)
Comment (T14Raptor) [inline server/block/slime.go:4]: This also also should implement `block.EntityLander`, and should reset the entity's fall distance if it implements `FallDistanceEntity`

**Review by T14Raptor — CHANGES_REQUESTED** [review]

**Comment by DaPigGuy** [comment]
Closing in favor of #445

**Pull request closed without merging**